### PR TITLE
Add detection by version for cve_2024_6385

### PR DESCRIPTION
**Pull request by PiranhaSa**

### CODE REVIEW

File: agent/definitions.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class VulnerabilityMetadata:
     title: str
     description: str
     reference: str
-    risk_rating: str
+    risk_rating: str = "CRITICAL"
```
Comment (mohsinenar) [inline agent/definitions.py:51]: We default risk rating to `CRITICAL`??

Comment (ErebusZ) [reply agent/definitions.py:51]: It was hardcoded in webexploit.WebExploit

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by deadly-panda — APPROVED** [review]

File: agent/definitions.py (hunk 1 above)
Comment (PiranhaSa) [reply agent/definitions.py:51]: yes that's what we said

**Review by nmasdoufi-ol — APPROVED** [review]

File: agent/exploits/cve_2024_6385.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "GitLab Authentication Bypass Vulnerability."
+VULNERABILITY_REFERENCE = "CVE-2024-6385"
+VULNERABILITY_DESCRIPTION = (
+    "An issue was discovered in GitLab CE/EE affecting all versions starting "
+    "from 15.8 prior to 16.11.6, starting from 17.0 prior to 17.0.4, "
+    "and starting from 17.1 prior to 17.1.2, which allows an attacker "
+    "to trigger a pipeline as another user under certain circumstances."
```
[the remark this answers is not in the document]
Comment (PiranhaSa) [reply agent/exploits/cve_2024_6385.py:14]: check it now please

**Review by 3asm — APPROVED** [review]

**Pull request merged**